### ISSUE TITLE
Can't use ConfigMap for /opt/flink/conf because entrypoint tries to write to it

### ISSUE THREAD

**Issue by rehevkor5** [lyft/flinkk8soperator#168]
It looks like you are relying on the ENTRYPOINT script to append the contents of OPERATOR_FLINK_CONFIG env var to `/opt/flink/conf/flink-conf.yaml`. Unfortunately, this makes it impossible to use a ConfigMap mounted as a volume to provide a default `flink-conf.yaml` and `log4j.properties`, because ConfigMap volumes are intentionally read-only. I can't see a way to make this happen.

Alternatively, you could provide a way for the Operator to control default values for checkpoint folders etc. so that state is always stored to a specific place (eg. S3 bucket) with the Job name as a component in the path (job name as one of the folder names). Or just generally allowing the operator to provide default flink-conf.yaml. An appropriate `log4j.properties` can be included in a Docker base image. But it would be nice for the operator to provide job-specific config.

**Comment by anandswaminathan** [comment]
@rehevkor5 This is not true. Operator merely sets the override values in `OPERATOR_FLINK_CONFIG `

What you are describing is the way our examples do. Since the image is passed in by the user. You have full control over how the image is built. So any form of default - state/zookeeper parameters can be configured as needed in your image.
